strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProviderTest.java

```diff
@@ -0,0 +1,238 @@
+package org.carlspring.strongbox.providers.repository;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.NoSuchAlgorithmException;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.xml.bind.JAXBException;
+
+import org.carlspring.strongbox.artifact.coordinates.NugetHierarchicalArtifactCoordinates;
+import org.carlspring.strongbox.artifact.generator.NugetPackageGenerator;
+import org.carlspring.strongbox.config.NugetLayoutProviderConfig;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.data.PropertyUtils;
+import org.carlspring.strongbox.providers.ProviderImplementationException;
+import org.carlspring.strongbox.services.ArtifactManagementService;
+import org.carlspring.strongbox.services.RepositoryManagementService;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryLayoutEnum;
+import org.carlspring.strongbox.storage.repository.RepositoryTypeEnum;
+import org.carlspring.strongbox.testing.TestCaseWithNugetPackageGeneration;
+import org.carlspring.strongbox.xml.configuration.repository.NugetRepositoryConfiguration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.carmatechnologies.commons.testing.logging.ExpectedLogs;
+import com.carmatechnologies.commons.testing.logging.api.LogLevel;
+
+import ru.aristar.jnuget.files.NugetFormatException;
+
+/**
+ * @author sbespalov
+ *
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = NugetLayoutProviderConfig.class)
+public class GroupRepositoryProviderTest
```

Comment:
I think it would be better to call this class `NugetGroupRepositoryProviderTest` and to rename the existing such test (for Maven) to `MavenGroupRepositoryProviderTest`. Otherwise, we'll have test cases with duplicate name...

Reply:
ok